pkg/client-sdk/client/client.go

```diff
@@ -24,56 +21,36 @@ var (
 	ErrConnectionClosedByServer = fmt.Errorf("connection closed by server")
 )
 
-type RoundEvent interface {
-	isRoundEvent()
+type BatchEvent interface {
```

Comment:
This interface sounds useless because we always cast the result and never use it. we could replace by `any` wherever it is used.